docs/distribution.rst

```diff
@@ -83,6 +86,36 @@ The test distribution algorithm is configured with the ``--dist`` command-line o
               pass
 
   This will make sure ``test1`` and ``TestA::test2`` will run in the same worker.
+
+  .. code-block:: python
+
+      @pytest.fixture(
+          scope="session",
+          params=[
+              pytest.param(
+                  "chrome",
+                  marks=pytest.mark.xdist_group("chrome"),
+              ),
+              pytest.param(
+                  "firefox",
+                  marks=pytest.mark.xdist_group("firefox"),
+              ),
+              pytest.param(
+                  "edge",
+                  marks=pytest.mark.xdist_group("edge"),
+              ),
+          ],
+      )
+      def setup_container():
+          pass
+
+
+      @pytest.mark.xdist_group(name="UIstuff")
+      def test1(setup_container):
+          pass
+
+  This will generate 3 new groups: ``chrome_UIstuff``, ``firefox_UIstuff`` and ``edge_UIstuff``.
```

Comment:
The example is correct, but I find it a bit misleading: one might at first glance think the order of the marks matter, because `UIstuff` is always appended at the end, but actually it ends up this way because `UI` sorts last.

How about:

```suggestion
      @pytest.mark.xdist_group(name="data_store")
      def test_data_store(setup_container):
          pass

  This will generate 3 new groups: ``chrome_data_store``, ``data_store_firefox`` and ``data_store`` (the markers are lexically sorted before being merged together).
```